pubsub/azurepubsub/example_test.go

```diff
@@ -131,3 +132,90 @@ func Example_openfromURL() {
 	s, err := pubsub.OpenSubscription(ctx, "azuresb://mytopic?subscription=mysub")
 	_, _, _ = t, s, err
 }
+
+func ExampleAckFuncForReceiveAndDelete() {
+
+	ctx := context.Background()
+	// See docs below on how to provision an Azure Service Bus Namespace and obtaining the connection string.
+	// https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-dotnet-get-started-with-queues
+	connString := os.Getenv("SERVICEBUS_CONNECTION_STRING")
+	topicName := "test-topic"
+	subscriptionName := "test-sub"
+
+	if connString == "" {
+		log.Fatal("Service Bus ConnectionString is not set")
+	}
+
+	// Construct a Service Bus Namespace from a SAS Token.
+	// See https://godoc.org/github.com/Azure/azure-service-bus-go#Namespace.
+	ns, err := azurepubsub.NewNamespaceFromConnectionString(connString)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Construct a Service Bus Topic for a topicName associated with a NameSpace.
+	// See https://godoc.org/github.com/Azure/azure-service-bus-go#Topic.
+	sbTopic, err := azurepubsub.NewTopic(ns, topicName, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer sbTopic.Close(ctx)
+
+	// Construct Receiver to AutoDelete messages.
+	// See https://godoc.org/github.com/Azure/azure-service-bus-go#SubscriptionWithReceiveAndDelete.
+	var opts []servicebus.SubscriptionOption
+	opts = append(opts, servicebus.SubscriptionWithReceiveAndDelete())
+
+	// Construct a Service Bus Subscription which is a child to a Service Bus Topic.
+	// See https://godoc.org/github.com/Azure/azure-service-bus-go#Topic.NewSubscription.
+	sbSub, err := azurepubsub.NewSubscription(sbTopic, subscriptionName, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer sbSub.Close(ctx)
+
+	// This package accommodates both kinds of systems. If your application uses
+	// at-least-once providers, it should always call Message.Ack. If your application
+	// only uses at-most-once providers (ReceiveAndDeleteMode), it may call Message.Ack, but does not need to.
+	//
+	// For more information on Service Bus ReceiveMode, see https://godoc.org/github.com/Azure/azure-service-bus-go#SubscriptionWithReceiveAndDelete.
+	//
+	// Use the options.AckOverride to supply your own logic for acking (optional regardless of modes). The custom AckOverride function can perform logging, no-ops,
+	// or your own bulk disposition operation. See https://godoc.org/github.com/Azure/azure-service-bus-go#Subscription.SendBatchDisposition.
+	//
+	// Important:
+	// 1. AckOverride is optional for Service Bus Subscription in ReceiveAndDeleteMode (SubscriptionWithReceiveAndDelete option enabled).
+	//    Calling the default driver Ack (opts.AckOverride is nil) in at-most-once mode will be disregarded by Azure Service Bus.
+	// 2. AckOverride can be used for custom batch dispositions (SubscriptionWithReceiveAndDelete option not set) but not recommended.
+	noop := func() {
+	}
+	subOpts := &azurepubsub.SubscriptionOptions{
+		AckFuncForReceiveAndDelete: noop,
```

Comment:
just inline the function body, on one line:
```
AckFuncForReceiveAndDelete: func(){},
```

Reply:
Ok